packages/scenario/src/jr/Scenario.ts

```diff
@@ -499,4 +501,31 @@ export class Scenario {
 	newInstance(): Promise<never> {
 		return Promise.reject(new UnclearApplicabilityError('Scenario instance statefulness'));
 	}
+
+	/**
+	 * @todo
+	 */
+	getValidationOutcome(): ValidateOutcome {
+		throw new ValidationImplementationPendingError();
+	}
+
+	/**
+	 * **PORTING NOTES**
+	 *
+	 * This currently deviates, intentionally, from JavaRosa interface, which
+	 * returns an instance of `FormIndex` (a concept not present in web forms, and
+	 * not currently anticipated). Since we already support JavaRosa's "event"
+	 * concepts, and since this is a reference lookup, we'll let that type be a
+	 * substitute for now.
+	 *
+	 * @todo This feels like a particular implementation detail we may want to
+	 * scrutinize after porting.
```

Comment:
I was going to update the comment to reflect this, but... I went and looked at all of the tests calling it and could not find one that seemed to have anything to do with either pages or TOC. Can you say more about what you mean here?

Reply:
"This" was not very clear! The `indexOf` method is only used a handful of times and I'll try to come up with a quick way to remove it. I think ideally we can get an XPath path for the form node that's being tested instead of using `indexOf` on the expected path and exposing JR innards.

Because `indexOf` returns a `FormIndex`, it got me thinking about the eventual need to have an engine/client interface concept of the current form index and jumping to a specific XPath path. It's more of a conceptual link than a practical one.